### PR TITLE
DOC-1889: restructure inline css documentation

### PR DESCRIPTION
**Pull request by brian-forte**
Ticket: DOC-1889: restructure inline css documentation

Changes:
* Restructured inline css documentation.

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] Changelog entry added
- [x] `modules/ROOT/nav.adoc` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/pages/inline-css.adoc [hunk 1]
```diff
@@ -37,55 +37,151 @@ tinymce.init({
...
+
+=== Inline mode support
+
+xref:use-tinymce-inline.adoc[Inline mode] is not sandboxed. Consequently, any specified CSS can effect how editor content presents. 
+
+CSS can be specified in inline mode in the following ways:
+
+. the `content_css` option
++
+Setting the `xref:add-css-options#content_css[content_css]` option automatically creates the `<link>` tag and puts it in the {productname} document’s `<head>`.
```
Comment (MitchCTiny) [inline modules/ROOT/pages/inline-css.adoc:117]:
```suggestion
Setting the xref:add-css-options.adoc#content_css[`content_css`] option automatically creates the `<link>` tag and puts it in the {productname} document’s `<head>`.
```

File: modules/ROOT/pages/inline-css.adoc [hunk 2]
```diff
@@ -37,55 +37,151 @@ tinymce.init({
...
+
+. `style` attributes applied to specific elements within a document. That is, inline CSS
+
+NOTE: By default, {productname} does not support `<style>` tags in the document `<body>`.
+
+
+=== Shadow DOM support
 
-CSS styling information can be specified in several places for iframe mode:
+IMPORTANT: Although the {pluginname} will likely work when an editor running in classic or inline mode is contained within a shadow root, Shadow DOM is **not** currently supported by {companynameformal}.
```
Comment (MitchCTiny) [inline modules/ROOT/pages/inline-css.adoc:165]:
```suggestion
IMPORTANT: Although the {pluginname} will likely work when an editor running in classic or inline mode is contained within a shadow root, Shadow DOM is **not** currently supported by {productname}.
```
I don't believe this is company wide.

File: modules/ROOT/pages/inline-css.adoc [hunk 3]
```diff
@@ -37,55 +37,151 @@ tinymce.init({
...
+When a {productname} editor in inline mode is contained within a shadow root, tags pointing to CSS declarations — either `<link>` tags pointing to stylesheet files or `<style>` tags containing specific CSS examples — are added as children of the shadow root. These tags do not need to be contained with the `<head>` tag.
 
-When in `inline mode`, the feature checks if the editor is inside a shadow root to ensure it does not try to get `<styles>` that are outside the shadow root.
+As well, when {productname} is working in inline mode, {pluginame} checks if the editor is inside a shadow root. If it finds it is, {pluginame} does not look for CSS declarations outside the shadow root.
 
-When an editor is in a shadow root, the feature should still support getting the stylesheets specified via the `content_css` option as well as the CSS styles specified via  the`content_style` option.
 
 // * some pseudo-class selectors. See the specification for more details
 // (https://notion.so/tinycloud/Technical-specification-7c16d4522d1241ba9e7364880eab3e51#94b1d13ff444492bbfce789f7876f39c)
```
Comment (MitchCTiny) [inline modules/ROOT/pages/inline-css.adoc:187]: Might be good to remove this line, the notion is intended to be internal only

File: modules/ROOT/pages/inline-css.adoc [hunk 4]
```diff
@@ -37,55 +37,151 @@ tinymce.init({
...
+[source,html]
+----
+button:hover { color: blue; }
+----
+
+. https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements[Pseudo-elements].
++
+For example:
++
+[source,html]
```
Comment (MitchCTiny) [inline modules/ROOT/pages/inline-css.adoc:53]:
```suggestion
[source,css]
```

File: modules/ROOT/pages/inline-css.adoc [hunk 5]
```diff
@@ -37,55 +37,151 @@ tinymce.init({
...
 The {pluginname} plugin does not support
 
-. https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes[Pseudo-classes] (for example `button:hover { color: blue; }`).
-. https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements[Pseudo-elements] (for example `p::first-line { color: blue; }`)
-. https://developer.mozilla.org/en-US/docs/Web/CSS/At-rule[@ rules] (for example `@media (screen)`)
+. https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes[Pseudo-classes].
++
+For example:
++
+[source,html]
```
Comment (MitchCTiny) [inline modules/ROOT/pages/inline-css.adoc:44]:
```suggestion
[source,css]
```

**Review by MitchCTiny — APPROVED** [review]
Thanks for cleaning this up 👍

File: modules/ROOT/pages/inline-css.adoc (hunk 3 above)
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc]: > Might be good to remove this line, the notion is intended to be internal only

Removed.

File: modules/ROOT/pages/inline-css.adoc [hunk 6]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
+----
+<head>
+<link rel="stylesheet" href="styles.css">
+</head>
+----
+
+. A `<link>` tag pointing to a stylesheet can also be manually inserted in the document `<head>`.
```
Comment (ltrouton) [inline modules/ROOT/pages/inline-css.adoc:136]: Equivalent to above point

File: modules/ROOT/pages/inline-css.adoc [hunk 7]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
+* CSS specified by the `xref:add-css-options.adoc#content_style[content_style]` option.
 
-When an `iframe` mode editor is contained with a shadow root, the editor content can be styled the same way, where the expected behavior for the `feature` is that it should not have to consider if the editor is in a shadow root. When an `inline` editor is contained within a shadow root, the stylesheet (`<link>` tags) and `<style>` tags are added directly as children of the shadow root and do not need to be contained within a `<head>` tag. 
 
-When in `inline mode`, the feature checks if the editor is inside a shadow root to ensure it does not try to get `<styles>` that are outside the shadow root.
+==== Classic mode
 
-When an editor is in a shadow root, the feature should still support getting the stylesheets specified via the `content_css` option as well as the CSS styles specified via  the`content_style` option.
+When a {productname} editor in classic mode is contained within a shadow root, CSS applied to the editor’s content will work as expected if the editor can also safely assume it is not operating in a shadow root.
```
Comment (ltrouton) [inline modules/ROOT/pages/inline-css.adoc:180]: Could just be me but I don't quite understand this sentence, mainly the 'if the editor can also safely assume it is not operating in a shadow root'.

File: modules/ROOT/pages/inline-css.adoc [hunk 8]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
-// see the tech spec for @-rule support:
-// https://notion.so/tinycloud/Technical-specification-7c16d4522d1241ba9e7364880eab3e51#555fede3a4ff4a26ad5efdf72bc1af6f.
 
-// This plugin allows extracting the content of the editor with CSS inlined.
-// Support classic and inline modes. Can be used in shadow dom. 
-// Should be able to extract CSS from the stylesheets defined via
-// `content_css` and `font_css` options, and CSS styles defined via `style_css` option.
+==== Inline mode
 
+When a {productname} editor in inline mode is contained within a shadow root, tags pointing to CSS declarations — either `<link>` tags pointing to stylesheet files or `<style>` tags containing specific CSS examples — are added as children of the shadow root. These tags do not need to be contained with the `<head>` tag.
```
Comment (ltrouton) [inline modules/ROOT/pages/inline-css.adoc:185]: I don't think this is quite accurate. TinyMCE will only automatically add styles specified via the `content_css` and `content_style` options into the shadow root. All other styles would have to be manually done by the integrator.

File: modules/ROOT/pages/inline-css.adoc [hunk 9]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
-*** The user can also manually insert a stylesheet link in the <head> of the page.
-* A CSS stylesheet that is specified in the `<body>` of the page i.e. `<link rel="stylesheet" href="styles.css">`
-** This is not recommended by https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link[MDN], but some browsers do allow it
-* A `<style>` in the `<head>` of the page i.e. `<style>p { color: red; }</style>`
-** To add a `<style>` in {productname} is by using the `content_style` option which automatically creates the `<style>` tag in the <head> of the document
-** The user can also manually put a `<style>` tag in the <head> of the page.
-* A `<style>` in the `<body>` of the page
-** As noted in the `Classic iframe` section, this should not be done, but some browsers will still allow it.
-* A `style` attribute on an {productname} element within the content i.e. inline CSS
+When a {productname} instance is in a shadow root, the {pluginame} should still support
```
[the remark this answers is not in the document]
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc:171]: To quote my commit message:

> Typo corrections: s/pluginame/pluginname/.

File: modules/ROOT/pages/inline-css.adoc [hunk 10]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
+. A CSS stylesheet specified in the `<head>` of the iframe.
++
+For example:
++
+[source,html]
+----
+<link rel="stylesheet" href="styles.css">
+----
+
+. A `<link>` tag pointing to a stylesheet can also be manually inserted in the iframe `<head>`.
```
[the remark this answers is not in the document]
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc:95]: Agreed. And combined.

File: modules/ROOT/pages/inline-css.adoc [hunk 11]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
+. A `<style>` tag in the `<head>` of the iframe.
++
+For example:
++
+[source,html]
+----
+<style>p { color: red; }</style>
+----
+
+. A `<style>` tag can also be manually inserted in the iframe `<head>`.
```
[the remark this answers is not in the document]
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc:106]: > Similar to my comment above, this point and the above point are equivalent.

Agreed.

> Also, it appears the new rework has lost mention of the content_style option

Lost mention is no longer lost. It is now a mirror of the `content_css` mention.

File: modules/ROOT/pages/inline-css.adoc [hunk 12]
```diff
@@ -37,75 +37,154 @@ tinymce.init({
...
+NOTE: By default, {productname} does not support `<style>` tags in the document `<body>`.
+
+
+=== Inline mode support
+
+xref:use-tinymce-inline.adoc[Inline mode] is not sandboxed. Consequently, any specified CSS can effect how editor content presents. 
+
+CSS can be specified in inline mode in the following ways:
+
+. the `content_css` option
```
[the remark this answers is not in the document]
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc:119]: Corrected upstream for the sake of simpler commits on my part.

File: modules/ROOT/pages/inline-css.adoc (hunk 6 above)
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc:136]: Agreed. And combined.

File: modules/ROOT/pages/inline-css.adoc (hunk 8 above)
Comment (brian-forte) [reply modules/ROOT/pages/inline-css.adoc:187]: The two sub-sections: Classic mode and Inline mode have been removed, replaced by a sub-clause in the re-written Shadow DOM support section.

File: modules/ROOT/pages/shadow-dom.adoc [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    // Note: use ‘target’ to specify the node the TinyMCE editor instance replaces.
+    // Do not use ‘selector’.
+    target: node,
+	plugins: [
+		"advlist autolink lists link image charmap print preview anchor",
+		"searchreplace visualblocks code fullscreen",
+		"insertdatetime media table help"
+	].join(' '),
+	toolbar: "insertfile undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image"
+  });
```
Comment (tiny-ben-tran) [inline modules/ROOT/pages/shadow-dom.adoc:39]:
```suggestion
  tinymce.init({
    // Note: use ‘target’ to specify the node the TinyMCE editor instance replaces.
    // Do not use ‘selector’.
    target: node,
    plugins: [
        "advlist autolink lists link image charmap print preview anchor",
        "searchreplace visualblocks code fullscreen",
        "insertdatetime media table help"
    ].join(' '),
    toolbar: "insertfile undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image"
  });
```

**Review by tiny-ben-tran — APPROVED** [review]

**Pull request merged**